vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinSavedRequestAwareAuthenticationSuccessHandler.java

```diff
@@ -123,17 +132,43 @@ public void onAuthenticationSuccess(HttpServletRequest request,
             throws ServletException, IOException {
         SavedRequest savedRequest = this.requestCache.getRequest(request,
                 response);
+        String storedServerNavigation = getStoredServerNavigation(request);
+        if (storedServerNavigation != null) {
+            response.setHeader(SAVED_URL_HEADER, storedServerNavigation);
```

Comment:
Setting the saved URL without the leading `/` would lead to an incorrect path when there is a context-root set for the application (see [this comment](https://github.com/vaadin/spring/pull/780#issuecomment-838295541)): 

```suggestion
            response.setHeader(SAVED_URL_HEADER, "/" + storedServerNavigation);
```

Maybe there is a better way to solved this, e.g. to change the stored URL instead, however, this solves the problem aparently.

Reply:
Fixed by prepending "/" when there is a context path